exporter/awsxrayexporter/translator/cause_test.go

```diff
@@ -74,7 +74,48 @@ func TestCauseWithHttpStatusMessage(t *testing.T) {
 	assert.True(t, strings.Contains(jsonStr, errorMsg))
 }
 
-func constructExceptionServerSpan(attributes map[string]interface{}) pdata.Span {
+func TestCauseWithZeroStatusMessage(t *testing.T) {
+	errorMsg := "this is a test"
+	attributes := make(map[string]interface{})
+	attributes[semconventions.AttributeHTTPMethod] = "POST"
+	attributes[semconventions.AttributeHTTPURL] = "https://api.example.com/widgets"
+	attributes[semconventions.AttributeHTTPStatusCode] = 500
+	attributes[semconventions.AttributeHTTPStatusText] = errorMsg
+
+	span := constructExceptionServerSpan(attributes, 0)
+	filtered, _ := makeHTTP(span)
+	// Status is used to determine whether an error or not.
+	// This span illustrates incorrect instrumentation,
+	// marking a success status with an error http status code, and status wins.
+	// We do not expect to see such spans in practice.
+	isError, isFault, filtered, cause := makeCause(span.Status(), filtered)
+
+	assert.False(t, isError)
+	assert.False(t, isFault)
+	assert.NotNil(t, filtered)
+	assert.Nil(t, cause)
+}
+
+func TestCauseWithClientErrorMessage(t *testing.T) {
+	errorMsg := "this is a test"
+	attributes := make(map[string]interface{})
+	attributes[semconventions.AttributeHTTPMethod] = "POST"
+	attributes[semconventions.AttributeHTTPURL] = "https://api.example.com/widgets"
+	attributes[semconventions.AttributeHTTPStatusCode] = 499
+	attributes[semconventions.AttributeHTTPStatusText] = errorMsg
+
+	span := constructExceptionServerSpan(attributes, 1)
+	filtered, _ := makeHTTP(span)
+
+	isError, isFault, filtered, cause := makeCause(span.Status(), filtered)
+
+	assert.True(t, isError)
+	assert.False(t, isFault)
+	assert.NotNil(t, filtered)
+	assert.NotNil(t, cause)
+}
+
+func constructExceptionServerSpan(attributes map[string]interface{}, statuscode int) pdata.Span {
```

Comment:
Instead of accepting int status code, make this parameter a type of pdata.StatusCode so it's more obvious what the code is in the tests (you will reference the constants)